### PR TITLE
os/arch/arm/src/amebasmart: I2S driver sampling fix

### PR DESCRIPTION
**Pull request by kaikaikaime**
Add a deinit function to allow register tx/rx direction for resampling
Add i2s rx

### CODE REVIEW

**Review by abhishek-samsung — COMMENTED** [review]
Hello, could you also please modify the functions ```i2s_txdatawidth``` and ```i2s_rxdatawidth```.  They not calling any amebasmart apis while changing datawidth.

https://github.com/Samsung/TizenRT/blob/bee6f97fb523726e509f1dbbda465e1e4e83c129/os/arch/arm/src/amebasmart/amebasmart_i2s.c#L580-L598

**Comment by kaikaikaime** [comment]
> Hello, could you also please modify the functions `i2s_txdatawidth` and `i2s_rxdatawidth`. They not calling any amebasmart apis while changing datawidth.
> 
> https://github.com/Samsung/TizenRT/blob/bee6f97fb523726e509f1dbbda465e1e4e83c129/os/arch/arm/src/amebasmart/amebasmart_i2s.c#L580-L598

Hi, I have pushed the modifications

File: os/arch/arm/src/amebasmart/amebasmart_i2s.c [hunk 1]
```diff
@@ -764,12 +764,37 @@ static int i2s_rxdma_prep(struct amebasmart_i2s_s *priv, struct amebasmart_buffe
...
+	if (NULL != bfcontainer && NULL != bfcontainer->apb) {
+
+		apb = bfcontainer->apb;
+		/* Add the container to the list of active DMAs */
+		sq_addlast((sq_entry_t *)bfcontainer, &priv->rx.act);
+
+		priv->apb_rx = apb; /* Reference current TX apb in our I2S struct */
+							/* Start sending first page, after that the txdma callback will be called in the tx irq handler */
+	}
+	i2s_recv_page(&priv->i2s_object, (u8 *) &priv->apb_rx->samp[priv->apb_rx->curbyte], I2S_DMA_PAGE_SIZE);
```
Comment (abhishek-samsung) [inline os/arch/arm/src/amebasmart/amebasmart_i2s.c:789]: I think this statement should be inside the if condition, could you please check?

**Review by abhishek-samsung — COMMENTED** [review]
datawidth related changes are not reflected in the current PR.

File: os/arch/arm/src/amebasmart/amebasmart_i2s.c (hunk 1 above)
Comment (kaikaikaime) [reply os/arch/arm/src/amebasmart/amebasmart_i2s.c]: Understood, datawidth modification and parameter should be in if condition

**Pull request closed without merging**